### ISSUE TITLE
Ability to send messages to threads

### ISSUE THREAD

**Issue by iguyking** [opsdroid/opsdroid#1229]
# Description
In Slack there is the ability to do threaded messaging.  It would be nice to be able to send a message into a thread from the original message.   

## Steps to Reproduce
Feature Enhancement

## Expected Functionality
Send a message and be able to say "attach it to this thread".

Possible approach:

```
async def send_message(self, message, thread=None):
```

## Experienced Functionality
Message is attached to the thread

## Versions
- **Opsdroid version:**  v0.16.0+77.g17247f9
- **Python version:** 3.7.3
- **OS/Docker version:**  Windows 10

## Configuration File
Please include your version of the configuration file below.

```yaml
# Your code goes here.

```

## Additional Details
Any other details you wish to include such as screenshots, console messages, etc.

**Comment by Cadair** [comment]
The API I would envisage for this is:

```python
connector.send(Message("my message", target="thread_ts"))
```

The question is how we easily expose the original thread_ts to users. It would currently require a connector specific trip into the `raw_event` attribute.

Maybe we should make connectors which support replies / threads to set the target of the event, so you could do:

```python
connector.send(Message("my message", target=message.target))
```

if fact, if we set `message.target` in the slack connector if the message is in a thread this:
```python
message.respond("Message")
```
would reply in the thread, because of [this](https://github.com/opsdroid/opsdroid/blob/master/opsdroid/events.py#L137) line.

So, in summary:

We should add a check for the existence of the `thread_ts` key in the slack event, and if it exists then set the `.target` attribute of the event the connector emits. This would therefore *by default* respond in the thread.

**Comment by jacobtomlinson** [comment]
I generally like this approach.

> This would therefore by default respond in the thread.

I'm not sure I like this. I use multiple Slacks on a day to day basis (hooray 😒) and each has its own etiquette. Some like threads, some hate threads. Therefore I think this should be configurable somehow.

Currently the target gets set to the room. Your proposal is to set the target to the `thread_ts` instead. Maybe we could have a configuration option like `respond-in-thread: false` to toggle this functionality?

Another option (which you might actually be suggesting) would be for the bot to respond to the room by default unless the user message is in a thread, in which case you respond in the thread.

The configuration option in that case could be whether you want the bot to start a thread by default or not.
